ui/models.py

```diff
@@ -42,45 +44,104 @@ class Video(models.Model):
+    """
+    A file associated with a Video object, either the original upload or a transcoded file.
+    """
+    created_at = models.DateTimeField(auto_now_add=True)
+    s3_object_key = models.TextField(unique=True)
+    bucket_name = models.CharField(max_length=128, blank=False, null=False)
+    video = models.ForeignKey(Video, on_delete=models.CASCADE)
+    encoding = models.CharField(max_length=128, default='original')
+    preset_id = models.CharField(max_length=128, blank=True, null=True)
 
     @property
     def s3_object(self):
-        """s3 object"""
-        bucket_name = os.environ.get("VIDEO_S3_BUCKET", "odl-video-service")
+        """
+        Get the S3 Object for the video
+
+            Returns:
+                s3.Object: Video file's S3 object
+        """
         s3 = boto3.resource('s3')
-        return s3.Object(bucket_name, self.s3_object_key)
+        return s3.Object(self.bucket_name, self.s3_object_key)
 
     @property
     def s3_url(self):
-        """s3 url"""
-        bucket_name = os.environ.get("VIDEO_S3_BUCKET", "odl-video-service")
-        encoded_key = urllib.parse.quote_plus(self.s3_object_key)
+        """
+        URL for the Video S3 object
+
+        Returns:
+            str: URL
+        """
         return "https://s3.amazonaws.com/{bucket}/{key}".format(
-            bucket=bucket_name,
-            key=encoded_key,
+            bucket=self.bucket_name,
+            key=self.s3_object_key,
         )
 
     @property
     def cloudfront_url(self):
-        """cloudfront url"""
+        """
+        Get the Cloudfront URL for the video
+
+        Returns:
+            str: Cloudfront unsigned URL
+        """
         distribution = os.environ.get("VIDEO_CLOUDFRONT_DIST")
         if not distribution:
             raise RuntimeError("Missing required env var: VIDEO_CLOUDFRONT_DIST")
-        encoded_key = urllib.parse.quote_plus(self.s3_object_key)
         return "https://{dist}.cloudfront.net/{key}".format(
             dist=distribution,
-            key=encoded_key,
+            key=self.s3_object_key
         )
 
+    @property
+    def cloudfront_presigned_url(self):
+        """
+        Get a signed Cloudfront URL with a default expiration date of 2 hours from
+        when this property is called.
+        """
+        expires = datetime.datetime.utcnow() + datetime.timedelta(hours=2)
+        return self.cloudfront_signed_url(expires)
+
     def cloudfront_signed_url(self, expires):
-        """cloudfront signed url"""
+        """
+        Get a signed Cloudfront URL for the video
+
+        Args:
+            expires (datetime): Expiration date for the link
+
+        Returns:
+            str: Cloudfront signed URL
+        """
         return make_cloudfront_signed_url(
             key=self.s3_object_key,
             expires=expires,
         )
 
     def __str__(self):
-        return self.title or "<untitled video>"
+        return self.s3_object_key
```

Comment:
not sure why you made the `__str__` so minimal and the `__repr__` that detailed, but I would make them pretty similar